### PR TITLE
Add Databricks private authentication shared service for SSO

### LINKED ISSUE #3182: Cannot login to Databricks - auth endpoint cannot be reached.
When trying to log into Databricks from a VM over private endpoint, I get the following:

image

Looks like a needed rule might be blocked.

cc @guybartal

### PR DESCRIPTION
**Pull request by guybartal**
# Resolves #3182 

## What is being addressed

Databricks SSO mechanism designed to work with a single Azure Databricks Auth sub resource deployed once per region and private DNS zone. This PR along with #3199 addresses that.

## How is this addressed

- Add a shared service to deploy a private authentication Databricks workspace with private link to Databricks DNS zone inside core.

### CODE REVIEW

File: templates/shared_services/databricks-auth/README.md [hunk 1]
```diff
@@ -0,0 +1,47 @@
+# Contents
```
Comment (marrobi) [inline templates/shared_services/databricks-auth/README.md:1]: Do we need this file?

File: templates/shared_services/databricks-auth/template_schema.json [hunk 2]
```diff
@@ -0,0 +1,31 @@
...
+      "title": "Name for the workspace service",
+      "description": "Azure Databricks Private Authentication Shared Service",
+      "default": "Azure Databricks Private Authentication Shared Service",
+      "updateable": true
+    },
+    "description": {
+      "type": "string",
+      "title": "Description of the workspace service",
+      "description": "Azure Databricks Private Authentication Shared Service",
+      "default": "Azure Databricks is a fast, easy, and collaborative Apache Spark-based big data analytics service designed for data science and data engineering.",
```
Comment (marrobi) [inline templates/shared_services/databricks-auth/template_schema.json:20]: Description should relate to what this enables.

File: templates/shared_services/databricks-auth/template_schema.json [hunk 3]
```diff
@@ -0,0 +1,31 @@
...
+      "updateable": true
+    },
+    "overview": {
+      "type": "string",
+      "title": "Azure Databricks Private Authentication Shared Service",
+      "description": "Long form description of the workspace service, in markdown syntax",
+      "default": "The Azure Databricks Lakehouse Platform provides a unified set of tools for building, deploying, sharing, and maintaining enterprise-grade data solutions at scale.\nAzure Databricks integrates with cloud storage and security in your cloud account, and manages and deploys cloud infrastructure on your behalf.\n[Azure Databricks documentation](https://learn.microsoft.com/en-us/azure/databricks/introduction/)",
```
Comment (marrobi) [inline templates/shared_services/databricks-auth/template_schema.json:27]: As above, this is auth specific.

File: templates/shared_services/databricks-auth/README.md (hunk 1 above)
Comment (guybartal) [reply templates/shared_services/databricks-auth/README.md]: removed

File: templates/shared_services/databricks-auth/template_schema.json (hunk 2 above)
Comment (guybartal) [reply templates/shared_services/databricks-auth/template_schema.json]: I've updated it to match AML bundle, but as a shared service, and provided default value which is related,
let me know if that's what you meant.

File: templates/shared_services/databricks-auth/template_schema.json (hunk 3 above)
Comment (guybartal) [reply templates/shared_services/databricks-auth/template_schema.json]: same as above

**Comment by marrobi** [comment]
Verified can access a DB workspace provisioned in a VNet outside of the TRE with a private endpoint in the shared services area:

image

**Comment by guybartal** [comment]
Thanks for testing that @marrobi!
I'll go ahead and merge it.

**Review by marrobi — APPROVED** [review]
Great work. @guybartal and @promisinganuj .

**Pull request merged**